mitodl/mit-open
# restrict learning path pages to learning path editors

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5208

### Description (What does it do?)
This PR wraps `LearningPathListingPage` and `LearningPathDetailsPage` in `RestrictedRoute`, requiring `Permissions.LearningPathEditor` to access them. `ListDetailsPage.test.tsx` was removed as it contains irrelevant tests that mostly checked to see if certain controls on the page were hidden based on permission level. Since these pages are now entirely restricted to Learning Path Editors, the tests are no longer necessary.

### How can this be tested?
 - Spin up `mit-learn` on this branch
 - Ensure that you have a normal user as well as as user with Learning Path Editor permission; either a user with `is_staff` or `is_superuser` set to true
 - Log in as the normal user and try and visit http://localhost:8063/learningpaths
 - Confirm that you are presented with a "Not allowed" error
 - Log in as your staff / super user and confirm that you can access the page
